odoo/addons/base/ir/ir_mail_server.py

```diff
@@ -23,6 +24,40 @@
 _test_logger = logging.getLogger('odoo.tests')
 
 
+def split_addr(addr, encoding):
+    """Split the address into local part and domain, properly encoded.
+       When non-ascii characters are present in the local part, it must be
+       MIME-word encoded. The domain name must be idna-encoded if it contains
+       non-ascii characters.
+    """
+    if '@' in addr:
+        localpart, domain = addr.split('@', 1)
+        try:
+            localpart.encode('ascii')
+        except UnicodeEncodeError:
+            localpart = Header(localpart, encoding).encode()
+        domain = domain.encode('idna').decode('ascii')
+    else:
+        localpart = Header(addr, encoding).encode()
+        domain = ''
+    return localpart, domain
+
+
+def sanitize_address(addr, encoding='utf-8'):
+    """Format a pair of (name, address) or an email address string."""
+    if not isinstance(addr, tuple):
+        addr = parseaddr(addr)
+    nm, addr = addr
+    try:
+        return formataddr((nm, addr))
```

Comment:
I would do: `return formataddr((Header(pycompat.to_text(nm)).encode(), addr))`
So we strictly keep the previous behavior.